### PR TITLE
Merge sparse to dense

### PR DESCRIPTION
**Pull request by martinjzhang**
1. Add scdrs.pp.preprocess for covariate correction and precomputing statistics.
2. Update scdrs.method.score_cell based on scdrs.pp.preprocess
3. Move reg_out, compute_stats, _get_mean_var to scdrs.pp
4. Update compute_score.py accordingly.
5. Files in scdrs/data/res/* are pushed because otherwise I cannot do the git rebase

### CODE REVIEW

File: scdrs/method.py [hunk 1]
```diff
@@ -468,7 +521,8 @@ def _correct_background(
             delimiter="\t",
         )
         np.savetxt(
-            save_intermediate + ".ctrl_raw_score.2nd_gs_alignment.tsv.gz",
+            save_intermediate
+            + ".ctrl_Re: MAGMA files for scDRSraw_score.2nd_gs_alignment.tsv.gz",
```
Comment (KangchengHou) [inline scdrs/method.py:525]: this is a typo

File: scdrs/data/res/toydata_gs_human.scdrs_ct.cell_type [hunk 2]
```diff
@@ -1,3 +1,3 @@
 	n_cell	n_ctrl	assoc_mcp	assoc_mcz	hetero_mcp	hetero_mcz
-causal_cell	10.0	20.0	0.04761905	12.297529	1.0	1.0
-non_causal_cell	20.0	20.0	0.9047619	-1.1364214	1.0	1.0
+causal_cell	10.0	20.0	0.04761905	11.596751	0.14285715	1.3146176
```
Comment (KangchengHou) [inline scdrs/data/res/toydata_gs_human.scdrs_ct.cell_type:2]: the change is due to the difference of dealing covariates between versions, is this correct?

**Review by KangchengHou — APPROVED** [review]

File: scdrs/data/res/toydata_gs_human.scdrs_ct.cell_type (hunk 2 above)
Comment (martinjzhang) [reply scdrs/data/res/toydata_gs_human.scdrs_ct.cell_type:2]: Yes.

File: scdrs/method.py (hunk 1 above)
Comment (martinjzhang) [reply scdrs/method.py:525]: Thanks, will update

**Pull request merged**